src/Serialization/basic_types.jl

```diff
@@ -9,6 +9,20 @@ function save_internal(s::SerializerState, z::fmpz)
 end
 
 
+################################################################################
+# fmpq
+function load_internal(s::DeserializerState, ::Type{fmpq}, q::Dict{Symbol, Int})
+    return fmpq(q)
```

Comment:
I don't think this will work (and there is no test for this code, which supports my theory). Perhaps you meant
```suggestion
    return fmpq(q[:num],q[:den])
```

Reply:
This is still not addressed, is it?

Reply:
right, I overlooked this, but seeing as my tests pass for polynomials with coefficients in QQ, seems like i dont need this code so it will be removed in my next commit

Reply:
Ah, indeed. It falls back to this generic code:
```julia
function save_internal(s::SerializerState, obj::T) where T
    result = Dict{Symbol, Any}()
    for (n,t) in zip(fieldnames(T), T.types)
        if n != :__attrs
            result[n] = save_type_dispatch(s, getfield(obj, n))
        end
    end
    return result
end

function load_internal(s::DeserializerState, ::Type{T}, dict::Dict) where T
    fields = []
    for (n,t) in zip(fieldnames(T), T.types)
        if n!= :__attrs
            push!(fields, load_type_dispatch(s, t, dict[n]))
        end
    end
    return T(fields...)
end
```

Reply:
this amounts to the same number of lines since fieldnames of fmpq is (:num, :den), so I think it makesense to leave as is? unless you think fmpq should have it's own de/serializer?

Reply:
it needs to have a dedicated one:
```
julia> a = fmpq(2, 3);
julia> a = a^100;
julia> a.den
4611686018472787752
```
just doing `getfield` is giving the wrong result. an `fmpq` is 2-fmpz. But as far as julia is concerned it is just 2-Ints. If the ints have the 2nd (?) highest bit NOT set, then the Int is the fmpz and your generic code works. Otherwise, it is a mangled pointer and your code will fail..(well return garbage)

Reply:
basically all Nemo types (types defined in c) need a dedicated implementation as the Julia struct does not reflect reality

Reply:
I see, so in my tests I was actually testing with Rational{Int64}, for some reason i thought 1//2 would give an fmpq, but it isn't, I will push some code now that handles fmpq types and modify a test for them, thanks for the clarification!